pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -107,19 +109,25 @@ func extractTopologyFromLabels(labels map[string]string) (map[string]string, str
 	}
 	// iterate in lowest to highest order as the lowest level should be sustained and higher level duplicate
 	// should be removed
-	duplicateTopology := make(map[string]int)
+	duplicateTopology := make(map[string][]string)
 	for i := len(allKubernetesTopologyLabels) - 1; i >= 0; i-- {
 		topologyLabel := allKubernetesTopologyLabels[i]
 		if value, ok := labels[topologyLabel]; ok {
 			if _, ok := duplicateTopology[value]; ok {
 				delete(topology, kubernetesTopologyLabelToCRUSHLabel(topologyLabel))
```

Comment:
Instead of keeping track of all duplicates for a single log message, how about if we just log each time a duplicate is found? Then this is a 1-line change. This is a rare condition, so would be nice to keep the code simpler. Your code does look good if we need to aggregate all the duplicates, but simpler is preferred when possible. 
```suggestion
				logger.Warningf(...)
				delete(topology, kubernetesTopologyLabelToCRUSHLabel(topologyLabel))
```

Reply:
Yea that'd be great. The only reason why I was passing this out is because this package (`topology`) doesn't currently have a logger instance, and I wasn't sure if this was intentional or not.

Reply:
Moved all the logic into this func. I kept the logic that collects a list of duplicate keys for each label, as I personally think it makes the logged output cleaner. Let me know if you'd prefer that this logs every duplicate value's key separately as soon as detected, as suggested above.

Reply:
Looks good to still aggregate to a single message